lib/Controller/WopiController.php

```diff
@@ -122,28 +120,64 @@ public function __construct(
 	 * and general information about the capabilities that the WOPI host has on the file.
 	 */
 	public function wopiCheckFileInfo(string $documentId): JSONResponse {
-		$token = $this->request->getParam('access_token');
+		$wopiToken = $this->request->getParam('access_token');
 
-		list($fileId, , $version, $sessionId) = Helper::parseDocumentId($documentId);
-		$this->logger->info('CheckFileInfo: Getting info about file {fileId}, version {version} by token {token}.', [
+		$this->logger->debug('CheckFileInfo: documentId {documentId}.', [
 			'app' => $this->appName,
-			'fileId' => $fileId,
-			'version' => $version,
-			'token' => $token ]);
-
-		$row = new Db\Wopi();
-		$row->loadBy('token', $token);
+			'documentId' => $documentId
+		]);
 
-		$res = $row->getWopiForToken($token);
+		$res = $this->getWopiInfoForToken($documentId, $wopiToken);
 		if (!$res) {
 			$this->logger->debug('CheckFileInfo: get token failed.', ['app' => $this->appName]);
-			return new JSONResponse([], Http::STATUS_NOT_FOUND);
+			return new JSONResponse([], Http::STATUS_FORBIDDEN);
+		}
+
+		// get origin
+		$postMessageOrigin = $res['server_host'];
+
+		// get owner info
+		$ownerId = $res['owner'];
+
+		// get user info
+		if ($res['editor'] && $res['editor'] !== '' && !($res['attributes'] & WOPI::ATTR_FEDERATED)) {
+			// file editing as local logged in user
+			$editor = $this->userManager->get($res['editor']);
+
+			$userId = $editor->getUID();
+			$userFriendlyName = $editor->getDisplayName();
+			$userEmail = $editor->getEMailAddress();
```

Comment:
It might be useful to know that the `$userEmail` could be set to null if the user doesn't have an email set.

Reply:
it is checked for null below.